schemas/com.snowplowanalytics.monitoring.batch/load_succeeded/jsonschema/2-0-1

```diff
@@ -0,0 +1,135 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a data load succeeding",
+  "self": {
+    "vendor": "com.snowplowanalytics.monitoring.batch",
+    "name": "load_succeeded",
+    "format": "jsonschema",
+    "version": "2-0-1"
+  },
+
+  "type": "object",
+  "properties": {
+    "shredding": {
+      "description": "Information about the batch taken from transformation step, isomorphic to shredding_complete schema",
+      "type": "object",
+      "properties": {
+        "base": {
+          "description": "Blob storage path to the root of the batch",
+          "type": "string",
+          "format": "uri",
+          "maxLength": 1024
+        },
+        "compression": {
+          "description": "File compression type",
+          "enum": ["GZIP", "NONE"]
+        },
+        "types": {
+          "description": "Set of Iglu URIs and the format they were shredded into",
+          "type": "array",
+          "items": {
+            "properties": {
+              "schemaKey": {
+                "description": "Iglu URI",
+                "type": "string",
+                "maxLength": 256
+              },
+              "format": {
+                "description": "File format in which the entities were shredded to",
+                "enum": ["TSV", "JSON", "WIDEROW", "PARQUET"]
```

Comment:
If we had to explain what each of these means:

- TSV.  Until now, we have used this to mean _shredded_ TSV.  In theory you could have non-shredded (wide) TSV, but I doubt we will ever do that.
- JSON.  Until now, we have used this to mean _shredded_ JSON.
- WIDEROW.  I assume you mean the output the of the transformer when we load into Snowflake?  So this is also JSON, but it is wide not shredded.
- PARQUET.  This format is also "wide", meaning a single parquet file contains all unstruct events and contexts.  In theory we could have shredded PARQUET - a separate file for each type - but I doubt we will ever do that.

I wonder if it's better to name the enum fields like this:
- TSV
- JSON
- WIDE_JSON
- WIDE_PARQUET

or even to name the format like this:
- TSV
- JSON
- PARQUET

...and then have a separate boolean field called "shredded" which is true for the old formats and false for our new formats.

Reply:
If we were planning to add new formats which outputs in `shredded` format as well, adding new "shredded" field would make sense completely. But since we are not planning to do that, I think it might create need for additional explanation for every new format in case we add `shredded` field. We need to explain new formats and `shredded = true` can not be together. Only `JSON` would be suitable for both `shredded = true` and `shredded = false` in that case.

IMO we might convert `WIDEROW` to `WIDE_JSON` but we can leave `PARQUET` as it is since we know new fields will only be in wide format. And specify that TSV and JSON are legacy formats.

Reply:
Now that I understand that `format` is a field within the `types` array, my opinion has changed.

I find it odd that we need to look _inside an array_ in order to discovery the format of the primary file.  It also permits nonsense JSONs, for example this is valid against the schema but it is nonsense:

```
types: [
  { schemaKey = abc, format = WIDE_JSON }
  { schemaKey = xyz, format = PARQUET
]
```

I wonder if we should say that a type's `format` is applicable only to shredded outputs.  If a batch was not shredded, then the items' formats are null.  (or set to `INTRINSIC`)

Then separately, we can have a `fileFormat` in the top level of the schema, to represent the format of the primary atomic event file.

Reply:
Even though it complicates the structure a bit, I guess it is better to do it like that. I didn't look at the what changes needs to be done in the code but I guess it won't be trivial. Just wanted to mention that.

Again pinging @chuwy in case if there is anything he wants to add.